src/coreclr/src/tools/ReadyToRun.SuperIlc/PathHelpers.cs

```diff
@@ -32,6 +32,9 @@ static class PathExtensions
 
     internal static string OSExeSuffix(this string path) => (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? path + ".exe" : path);
 
+    internal static string OSDllSuffix(this string path) => path +
+        (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? ".dll" : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? ".dylib" : ".so");
```

Comment:
This might be considered a pre-existed issue, but I would prefer naming these functions `AppendOSDllSuffix` or `Add...`, because they append the suffix to the argument.

Reply:
Good point, I've also been thinking that I originally named them stupidly :-).

Reply:
Fixed in 2nd commit.

Reply:
Thank you, looks good!